File: code/python-for-csharp-java/compare/L02Values.java
// compare/L02Values.java
class L02Values {
    public static void main(String[] args) {
        // Autoboxing goes through Integer.valueOf, which caches the values from -128 to 127
        Integer small = 100, alsoSmall = 100;
        Integer big = 1000, alsoBig = 1000;
        System.out.println((small == alsoSmall) + " " + small.equals(alsoSmall));
        System.out.println((big == alsoBig) + " " + big.equals(alsoBig));

        // The compiler converts the int to text
        int quantity = 3;
        System.out.println("quantity: " + quantity);
    }
}

File: code/python-for-csharp-java/compare/l02_values.cs
// compare/l02_values.cs
// Boxing an int creates a new object every time: == on object compares references
object small = 100, alsoSmall = 100;
object big = 1000, alsoBig = 1000;
Console.WriteLine($"{small == alsoSmall} {small.Equals(alsoSmall)}");
Console.WriteLine($"{big == alsoBig} {big.Equals(alsoBig)}");

// The compiler converts the int to text
int quantity = 3;
Console.WriteLine("quantity: " + quantity);
